### ISSUE TITLE
SharePayload.Commit() never completes when AddMedia is given an image path (on Android)

### ISSUE THREAD

**Issue by mattdw** [natmlx/NatShare#131]
The following code works perfectly when given a path to a video (ending in .mp4), and in all cases on iOS, but not when given a path to an image on Android. No error is logged or exception thrown; the task just never completes.

The device is an OPPO A54 running Android 12, but we saw it on multiple Android devices.

```c#
var payload = new SharePayload();
payload.AddMedia(path);
var res = await payload.Commit();
```

✅ iOS *.mp4
✅ iOS *.png
✅ Android *.mp4
❌ Android *.png
❌ Android *.PNG
❌ Android *.jpg
❌ Android *.jpeg

My current workaround is to use the `AddImage` method; **this seems to work**:

```c#
var payload = new SharePayload();

Texture2D tex = new Texture2D(2, 2);
tex.LoadImage(System.IO.File.ReadAllBytes(path));
payload.AddImage(tex);

var res = await payload.Commit();
```

**Comment by olokobayusuf** [comment]
Hey @mattdw thanks for reporting this. Checking it out for the 1.2.7 release.

**Comment by mattdw** [comment]
There's a possibility this was a permissions problem with `Application.temporaryCachePath`, we had different (partially better) results placing the image in `Application.persistentDataPath`.

**Comment by olokobayusuf** [comment]
[Fixed in NatShare 1.3](https://github.com/natmlx/NatShare/releases/tag/1.3.0).
